src/Explorer/Controls/Settings/SettingsComponent.tsx

```diff
@@ -0,0 +1,1285 @@
+      const label = "Discard";
+      buttons.push({
+        iconSrc: DiscardIcon,
+        iconAlt: label,
+        onCommandClick: this.onRevertClick,
+        commandButtonLabel: label,
+        ariaLabel: label,
+        hasPopup: false,
+        disabled: !this.discardSettingsChangesButton.isEnabled()
+      });
+    }
+    return buttons;
+  };
+
+  private onMaxAutoPilotThroughputChange = (newThroughput: number): void =>
+    this.setState({ autoPilotThroughput: newThroughput });
+
+  private onThroughputChange = (newThroughput: number): void => this.setState({ throughput: newThroughput });
+
+  private onAutoPilotSelected = (isAutoPilotSelected: boolean): void =>
+    this.setState({ isAutoPilotSelected: isAutoPilotSelected });
+
+  private onAutoPilotTierChange = (selectedAutoPilotTier: DataModels.AutopilotTier): void =>
+    this.setState({ selectedAutoPilotTier: selectedAutoPilotTier });
+
+  private getStatusBarComponent = (): JSX.Element => (
+    <div className="warningErrorContainer scaleWarningContainer">
+      {this.shouldShowNotificationStatusPrompt() && (
+        <div className="warningErrorContent">
+          <span>
+            <Image src={InfoColor} alt="Info" />
+          </span>
+          <div className="warningErrorDetailsLinkContainer">{this.state.notificationStatusInfo}</div>
+        </div>
+      )}
+      {!this.shouldShowNotificationStatusPrompt() && (
+        <div className="warningErrorContent">
+          <span>
+            <Image src={Warning} alt="Warning" />
+          </span>
+          <div className="warningErrorDetailsLinkContainer">
+            {!!this.getWarningMessage() && this.getWarningMessage()}
+          </div>
+        </div>
+      )}
+    </div>
+  );
+
+  private onPivotChange = (item: PivotItem): void => {
+    const selectedTab = SettingsV2TabTypes[item.props.itemKey as keyof typeof SettingsV2TabTypes];
+    this.setState({ selectedTab: selectedTab });
+  };
+
+  public render(): JSX.Element {
+    const tabs: SettingsV2TabInfo[] = [];
+    if (!hasDatabaseSharedThroughput(this.collection)) {
+      tabs.push({
+        tab: SettingsV2TabTypes.ScaleTab,
+        content: (
+          <ScaleComponent
+            collection={this.collection}
+            container={this.container}
+            hasProvisioningTypeChanged={this.hasProvisioningTypeChanged}
+            hasAutoPilotV2FeatureFlag={this.hasAutoPilotV2FeatureFlag}
+            isFixedContainer={this.isFixedContainer}
+            autoPilotTiersList={this.autoPilotTiersList}
+            onThroughputChange={this.onThroughputChange}
+            throughput={this.state.throughput}
+            throughputBaseline={this.state.throughputBaseline}
+            autoPilotThroughput={this.state.autoPilotThroughput}
+            autoPilotThroughputBaseline={this.state.autoPilotThroughputBaseline}
+            selectedAutoPilotTier={this.state.selectedAutoPilotTier}
+            isAutoPilotSelected={this.state.isAutoPilotSelected}
```

Comment:
Instead of passing so many props, can you pass on scale prop object which has these values?

Reply:
I dont think that will streamline it much. a ScaleComponentProps interface already exists and these props being passed are a part of that.